doc/adrs/192_add_configurable_render_variables_to_tree.adoc

```diff
@@ -0,0 +1,114 @@
+VariableManager renderVariables = this.treeDescription.getRenderVariablesProvider().apply(this.variableManager);
+for (Map.Entry<String, Object> renderVariable : renderVariables.getVariables().entrySet()) {
+    String name = renderVariable.getKey();
+    Object value = renderVariable.getValue();
+    if (!this.variableManager.hasVariable(name)) {
+        this.variableManager.put(name, value);
+    }
+}
+----
+
+
+A specific tree description can use this to enrich the evaluation context with additional information it knows will be needed.
+
+For example, the definition if `ExplorerDescriptionProvider` can add:
+
+[source,java]
+----
+@Override
+public List<IRepresentationDescription> getRepresentationDescriptions(IEditingContext editingContext) {
+    var explorerTreeDescription = TreeDescription.newTreeDescription(DESCRIPTION_ID)
+         // ...
+         .renderVariablesProvider(this::getRenderVariables)
+         .build();
+    return List.of(explorerTreeDescription);
+}
+
+private VariableManager getRenderVariables(VariableManager variableManager) {
+    var result = new VariableManager();
+    var optionalEditingContext = variableManager.get(IEditingContext.EDITING_CONTEXT, IEditingContext.class);
+    if (optionalEditingContext.isPresent()) {
+        String editingContextId = optionalEditingContext.get().getId();
+        var optionalSemanticDataId = new UUIDParser().parse(editingContextId);
+        List<RepresentationMetadata> allRepresentationMetadata = this.representationMetadataSearchService.findAllRepresentationMetadataBySemanticData(AggregateReference.to(optionalSemanticDataId.get()));
+        result.put("existingRepresentations", allRepresentationMetadata);
+    }
+    return result;
+}
+----
+
+Which, with *one* access to the database, contributes a new variable `existingRepresentations`.
+
+`ExplorerChildrenProvider.hasChildren` can then be rewritten to leverage this new variable:
+
+[source,java]
+----
+@Override
+public boolean hasChildren(VariableManager variableManager) {
+    Object self = variableManager.getVariables().get(VariableManager.SELF);
+    Optional<IEditingContext> optionalEditingContext = variableManager.get(IEditingContext.EDITING_CONTEXT, IEditingContext.class);
+    List<RepresentationMetadata> existingRepresentations = variableManager.get("existingRepresentations", List.class).orElse(List.of());
+    return this.explorerServices.hasChildren(self, optionalEditingContext.orElse(null), existingRepresentations);
+}
+----
+
+With this change, the hundreds/thousands of individual DB queries currently needed to answer `hasChildren` for each leaf node (see issue #4202) are replaced with a single query at the beginning of the explorer render.
+
+=== Cutting backs
+
+=== Rabbit holes
+
+== Status
+
+Proposed
+
+== Consequences
+
+* If we implement this in `Tree` (notably to improve the _Explorer_) and it proves useful, the same mechanism could be generalized to all representations.
```

Comment:
After some digging I remember that we did it this way for forms already and it was a bit limited